app/models/custom_button_set.rb

```diff
@@ -1,6 +1,19 @@
 class CustomButtonSet < ApplicationRecord
   acts_as_miq_set
 
+  before_save :update_children
+  def update_children
+    if set_data.nil? || set_data[:button_order].nil?
```

Comment:
I think this could be a one-liner and you don't need the `OR` at all if *"you gotta get up and `try`, and `try`, and `try`"* :rofl:
```suggestion
    return remove_all_children if set_data.try(:[], :button_order).nil?
```

Reply:
Can't be a one-liner, if `remove_all_children` ever returns false, it will abort in Rails 4.

But :+1: , updating to use `try`.